edge-functions/serverside-analytics/README.md

```diff
@@ -0,0 +1,58 @@
+# Serverside Analytics
+
+This example shows how to use [Segment](https://segment.com) to implement server-side analytics at the edge with [Next.js Middleware](https://nextjs.org/docs/middleware).
+
+## Demo
+
+https://edge-functions-serverside-analytics.vercel.app
+
+## How to Use
+
+You can choose from one of the following two methods to use this repository:
+
+### One-Click Deploy
+
+Deploy the example using [Vercel](https://vercel.com?utm_source=github&utm_medium=readme&utm_campaign=next-example):
+
+[![Deploy with Vercel](https://vercel.com/button)](https://vercel.com/new/git/external?repository-url=https://github.com/vercel/examples/tree/main/edge-functions/serverside-analytics&project-name=serverside-analytics&repository-name=serverside-analytics&env=SEGMENT_WRITE_KEY&envDescription=A%20Segment%20write%20key&envLink=https%3A%2F%2Fgithub.com%2Fvercel%2Fexamples%2Ftree%2Ffeat%2Fserverside-analytics%2Fedge-functions%2Fserverside-analytics%23setting-up-segment)
+
+### Clone and Deploy
+
+Execute [`create-next-app`](https://github.com/vercel/next.js/tree/canary/packages/create-next-app) with [npm](https://docs.npmjs.com/cli/init) or [Yarn](https://yarnpkg.com/lang/en/docs/cli/create/) to bootstrap the example:
+
+```bash
+npx create-next-app --example https://github.com/vercel/examples/tree/main/edge-functions/serverside-analytics
+# or
+yarn create next-app --example https://github.com/vercel/examples/tree/main/edge-functions/serverside-analytics
+
+```
+
+### Setting up Segment
+
+1. Register and create a workspace at [segment.com](https://segment.com)
+2. Add a [javascript source](https://segment.com/docs/connections/sources/catalog/libraries/website/javascript/) and grab the `writeKey`
```

Comment:
@gbibeaul I did not understand what todo after clicking that link, let's add more detailed instructions in this step.